packages/components/Button/src/Button.types.ts

```diff
@@ -148,6 +154,7 @@ export interface ButtonSlotProps {
   root: React.PropsWithRef<IViewProps>;
   icon: IconProps;
   content: TextProps;
+  shadow?: ShadowProps;
```

Comment:
Is the plan to add a shadow slot to every button?

Reply:
At least for macOS, we are meeting later today to discuss which variants should have shadows by default. For the final implementation, I have two ideas on this bit:

1) Create a Shadow slot that we optionally render if we have a shadow token specified. This will always add 2 extra views to the hierarchy. This is more or less what this PR is doing. We scoped it to FAB here since that Button will _always_ have a shadow. 

2) Create a "RootWithShadow" component where the shadow slots' innerView becomes the new root, and the "withShadow" bit is in charge of rendering a key and ambient shadow. This would only add 1 more view to the hierarchy, since it would take the place of the root slot. The issue is we would have to be more careful about making sure props are passed properly. Also.. we might then want a "PressableWithShadow" too if we do that.

Reply:
For option 2 you could do something like what I do with the MenuTrigger's child, cloning the child element and adding the props needed to create a shadow.

Reply:
Added a task for myself to investigate this later, thanks for the suggestions!